src/Components/EditMovieFormik/EditMovieFormik.tsx

```diff
@@ -0,0 +1,83 @@
+import { useId } from 'react';
+import { Formik, Form, FormikProps, FormikHelpers } from 'formik';
+import { genres } from '../../containers/MoviesListOptionsContainer/genres';
+import { EditMovieFormValue } from '../../models/EditMovieFormValue';
+import { Genre } from '../../models/Genre';
+import { Movie } from '../../models/Movie';
+import { TextField } from '../TextField/TextField';
+import { FormSelect } from '../FormSelect/FormSelect';
+import { validationSchema } from './validationSchema';
+import { createMovie, editMovie, fetchMovies } from '../../store/moviesReducer';
+import { useAppDispatch } from '../../hooks/useAppDispatch';
+import { useMovies } from '../../hooks/useMovies';
+import { getMovieFromFormValue } from '../../utils/getMovieFromFormValue';
+import { AsyncSubmitStatus } from '../../models/AsyncSubmitStatus';
+import './EditMovie.scss';
+
+interface EditMovieProps {
+  movie: Movie | null;
+  handleClose: () => void;
+}
+
+function setInitialFormValue(movie: Movie | null): EditMovieFormValue {
+  return {
+    title: movie?.title || '',
+    release_date: movie?.release_date || '',
+    poster_path: movie?.poster_path || '',
+    vote_average: movie?.vote_average?.toString() || '',
+    runtime: movie?.runtime?.toString() || '',
+    overview: movie?.overview || '',
+    genres: movie?.genres.map((value: string): Genre => ({ value, label: value })) || [],
+  };
+}
+
+export const EditMovieFormik = ({ movie, handleClose }: EditMovieProps) => {
+  const dispatch = useAppDispatch();
+  const { queryParams } = useMovies();
+
+  const inputIdPrefix = useId();
+  const getIdFor = (fieldName: string): string => `${inputIdPrefix}_${fieldName}`;
+
+  const handleFormSubmit = async (formValue: EditMovieFormValue, { setSubmitting, setStatus }: FormikHelpers<EditMovieFormValue>) => {
+    const isEditing = !!movie?.id;
+
+    const formMovie: Partial<Movie> = getMovieFromFormValue(formValue);
+    const actionToDispatch = isEditing ? editMovie({ ...formMovie, id: movie.id }) : createMovie(formMovie);
+
+    try {
+      await dispatch(actionToDispatch).unwrap();
+      dispatch(fetchMovies(queryParams));
+      handleClose();
+      setStatus(AsyncSubmitStatus.SUBMIT_SUCCESS);
+    } catch (rejectedValueOrSerializedError) {
+      setStatus(AsyncSubmitStatus.SUBMIT_FAIL);
+    } finally {
+      setSubmitting(false);
+    }
+  };
+
+  return (
+    <Formik initialValues={setInitialFormValue(movie)} validationSchema={validationSchema} onSubmit={handleFormSubmit}>
+      {({ isSubmitting, status }: FormikProps<EditMovieFormValue>) => (
```

Comment:
Render prop pattern is always rerendered. So, I would use useCallback here or use formik context. It is just note for the feature